rafthttp/snapshot_sender.go

```diff
@@ -104,7 +105,9 @@ func (s *snapshotSender) send(merged snap.Message) {
 // post posts the given request.
 // It returns nil when request is sent out and processed successfully.
 func (s *snapshotSender) post(req *http.Request) (err error) {
-	cancel := httputil.RequestCanceler(req)
+	ctx, cancel := context.WithCancel(context.Background())
+	req = req.WithContext(ctx)
+	defer cancel()
```

Comment:
@xiang90 i had to do this change because the govet in CI was giving a possible go routine leak error with old cancel placed there since it expects cancel to be called mandatorily before exiting. What do you want me to do for this?